### PR TITLE
Issues 10, 91 StreamInfo request changes to retrieve subjects, deleted details

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/main/java/io/nats/client/JetStreamManagement.java [hunk 1]
```diff
@@ -74,6 +74,17 @@
...
+    /**
+     * Gets the info for an existing stream, and include subject info.
+     * @param streamName the stream name to use.
+     * @param subjectFilter the subject filter to use. Can be wildcard. Null is equivalent to all subjects, the same as '>'
+     * @throws IOException covers various communication issues with the NATS
+     *         server such as timeout or interruption
+     * @throws JetStreamApiException the request had an error related to the data
+     * @return stream information
+     */
+    StreamInfo getStreamInfoWithSubjectInfo(String streamName, String subjectFilter) throws IOException, JetStreamApiException;
```
Comment (scottf) [inline src/main/java/io/nats/client/JetStreamManagement.java:86]: looking for suggestions for a better name for this. I didn't want to overload getStreamInfo b/c the behavior is different, namely the original does not get any subject info

File: src/main/java/io/nats/client/api/StreamInfoOptions.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 The NATS Authors
```
Comment (kozlovic) [inline src/main/java/io/nats/client/api/StreamInfoOptions.java:1]: Date: 2022

File: src/main/java/io/nats/client/api/Subject.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 The NATS Authors
```
Comment (kozlovic) [inline src/main/java/io/nats/client/api/Subject.java:1]: Date: 2022

File: src/main/java/io/nats/client/api/StreamInfoOptions.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public String toJson() {
+        StringBuilder sb = beginJson();
+        addField(sb, SUBJECTS_FILTER, subjectsFilter);
```
Comment (kozlovic) [inline src/main/java/io/nats/client/api/StreamInfoOptions.java:66]: This JSON field should be added ONLY when the subject filter is actually specified. Is addField() doing that under the cover?

File: src/test/java/io/nats/client/impl/JetStreamManagementTests.java [hunk 5]
```diff
@@ -236,9 +239,55 @@ public void testGetStreamInfo() throws Exception {
...
+                PublishAck pa = jsPublish(js, subject(x), data(x + 2));
+                packs.add(pa);
+                jsm.deleteMessage(STREAM, pa.getSeqno());
+            }
+            jsPublish(js, "foo.bar", 6);
+
             StreamInfo si = jsm.getStreamInfo(STREAM);
             assertEquals(STREAM, si.getConfiguration().getName());
+            assertEquals(6, si.getStreamState().getSubjectCount());
+            assertNull(si.getStreamState().getSubjects());
```
Comment (kozlovic) [inline src/test/java/io/nats/client/impl/JetStreamManagementTests.java:263]: Ok, that seem to indicate that getStreamInfo() without the option is getting no list at all.

**Review by kozlovic — CHANGES_REQUESTED** [review]
Little change to the date for the new file's header. Also, asking for a confirmation about addField() to make sure that nothing gets added if param is "empty/null" string.

File: src/main/java/io/nats/client/api/StreamInfoOptions.java (hunk 4 above)
Comment (scottf) [reply src/main/java/io/nats/client/api/StreamInfoOptions.java:66]: yes, addField checks for null/empty and doesn't add it.

File: src/test/java/io/nats/client/impl/JetStreamManagementTests.java (hunk 5 above)
Comment (scottf) [reply src/test/java/io/nats/client/impl/JetStreamManagementTests.java:263]: I could have given an empty list though, but null is more consistent with the rest of the api.

File: src/main/java/io/nats/client/api/Subject.java (hunk 3 above)
Comment (scottf) [reply src/main/java/io/nats/client/api/Subject.java]: fixed

**Review by kozlovic — APPROVED** [review]

**Pull request merged**